kaspersky.py

```
@@ -53,7 +54,11 @@ def start(self) -> None:
 
     def execute(self, request: ServiceRequest) -> None:
         request.result = Result()
-        icap_result = self.icap.scan_data(request.file_contents, request.file_name)
+        try:
+            icap_result = self.icap.scan_data(request.file_contents, request.file_name)
+        except timeout:
+            # Try again, but only once!
+            icap_result = self.icap.scan_data(request.file_contents, request.file_name)
```

Let it raise the error if it fails/times out twice?

---

Yes, similar to how MD handles connection errors (https://github.com/CybercentreCanada/assemblyline-service-metadefender/blob/992a4cc60a57dfcde353bcc93e26ea47d24b8c27/metadefender.py#L233), we could just raise a Recoverable error but then that has the potential to get stuff in an infinite loop

---

Could adopt a "three strikes, you're out" approach by encapsulating inside a for and break after icap_result; exception could be caught and logged or passed and lets the loop continue until it completes?

---

sure! good thinking!